app/code/Magento/InventoryReservations/Model/GetOrderWithBrokenReservation.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryReservations\Model;
+
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Model\ResourceModel\Order\Collection;
+
+class GetOrderWithBrokenReservation
+{
+    /**
+     * @var GetOrderInFinalState
+     */
+    private $getOrderInFinalState;
+    /**
+     * @var GetReservationsList
+     */
+    private $getReservationsList;
+    /**
+     * @var Json
+     */
+    private $json;
+
+    /**
+     * GetOrderWithBrokenReservation constructor.
+     * @param GetOrderInFinalState $getOrderInFinalState
+     * @param GetReservationsList $getReservationsList
+     * @param Json $json
+     */
+    public function __construct(
+        GetOrderInFinalState $getOrderInFinalState,
+        GetReservationsList $getReservationsList,
+        Json $json
+    ) {
+        $this->getOrderInFinalState = $getOrderInFinalState;
+        $this->getReservationsList = $getReservationsList;
+        $this->json = $json;
```

Comment:
Using `Json` is ok, but I am wondering if using `SerializeInterface` is a better option.